fili-core/src/main/java/com/yahoo/bard/webservice/web/filters/RoleBasedAuthFilter.java

```diff
@@ -50,17 +51,53 @@ public RoleBasedAuthFilter(ObjectMapper mapper) {
     @Override
     @SuppressWarnings("checkstyle:cyclomaticcomplexity")
     public void filter(ContainerRequestContext containerRequestContext) throws IOException {
-        List<String> allowedUserRoles = getAllowedUserRoles();
-        if (!allowedUserRoles.isEmpty() && !isUserInRole(allowedUserRoles, containerRequestContext)) {
-                abortRequest(
-                        containerRequestContext,
-                        Response.Status.FORBIDDEN,
-                        BouncerAuthorizationStatus.ACCESS_DENIED
-                );
+
+        if (!isAuthorized(containerRequestContext)) {
+            abortRequest(
+                    containerRequestContext,
+                    Response.Status.FORBIDDEN,
+                    BouncerAuthorizationStatus.ACCESS_DENIED
+            );
         }
     }
 
     /**
+     * Request authorization logic, bypassing OPTIONS request and authorize if user is in authorized role or
+     * authorized role is not specified.
+     *
+     * @param containerRequestContext containing the request context information
+     *
+     * @return true if request is authorized, false otherwise
+     */
+    private boolean isAuthorized(ContainerRequestContext containerRequestContext) {
+        return isBypassMethod(containerRequestContext) || isUserInRole(getAllowedUserRoles(), containerRequestContext);
```

Comment:
I was thinking we should just make it a bug issue and fix it later on, because it should be obvious that this behavior is undesirable.